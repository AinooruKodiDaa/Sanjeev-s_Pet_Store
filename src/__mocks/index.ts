export * from "./mockRouter"
export * from "./redux"