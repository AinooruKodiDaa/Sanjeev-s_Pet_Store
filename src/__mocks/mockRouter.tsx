import React, { PropsWithChildren } from 'react';
import { MemoryRouter } from 'react-router-dom';

export const mockNavigate = jest.fn();

const MockRouter: React.FC<PropsWithChildren> = ({ children }) => (
  <MemoryRouter>
    {React.cloneElement(children as React.ReactElement, {
      navigate: mockNavigate,
    })}
  </MemoryRouter>
);

export default MockRouter;
