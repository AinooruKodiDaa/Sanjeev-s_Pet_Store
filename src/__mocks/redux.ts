import { AnyAction, Observable, Reducer, Store } from "redux";

export const initialState = {};

export const mockDispatch = jest.fn();

export const mockStore: Store<any, AnyAction> = {
  getState: jest.fn(() => initialState),
  dispatch: mockDispatch,
  subscribe: jest.fn(),
  replaceReducer: function (nextReducer: Reducer<any, AnyAction, any>): void {
    throw new Error("Function is not implemented");
  },
  [Symbol.observable]: function (): Observable<any> {
    throw new Error("Function not implemented");
  },
};
