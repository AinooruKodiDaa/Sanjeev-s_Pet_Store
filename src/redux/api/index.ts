export* from "./petsApi"
export* from "./statesApi"
export* from "./categoriesApi"
export* from "./usersApi"