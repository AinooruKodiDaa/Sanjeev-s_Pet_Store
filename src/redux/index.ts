import store from './store/configureStore';

export default store;
