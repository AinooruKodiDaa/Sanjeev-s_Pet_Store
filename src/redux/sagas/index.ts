import { StatesState } from "../types/statesType";
import { all } from "redux-saga/effects";
import authSaga from "./authSaga";
import usersSaga from "./usersSaga";
import petsSaga from "./petsSaga";
import statesSaga from "./statesSaga";
import categoriesSaga from "./categoriesSaga";


function* rootSaga() {
  yield all([
    authSaga(),
    usersSaga(),
    petsSaga(),
    statesSaga(),
    categoriesSaga()
    
    // Add other sagas here if you have more
  ]);
}

export default rootSaga;
