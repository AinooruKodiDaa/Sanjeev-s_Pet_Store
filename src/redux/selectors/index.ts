export * from "./petsSelector"
export * from "./statesSelector"
export * from "./categoriesSelector"