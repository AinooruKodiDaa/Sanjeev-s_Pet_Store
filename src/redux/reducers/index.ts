import { combineReducers } from "redux";
import usersReducer from "./usersReducer";
import authReducer from "./authReducer";
import petsReducer from "./petsReducer";
import statesReducer from "./statesReducer";
import categoriesReducer from "./categoriesReducer";

const rootReducer = combineReducers({
  auth: authReducer,
  users: usersReducer,
  pets: petsReducer,
  states: statesReducer,
  categories: categoriesReducer
});

export default rootReducer;
