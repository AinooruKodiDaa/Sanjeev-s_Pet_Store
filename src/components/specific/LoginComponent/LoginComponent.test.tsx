import React from "react";
import { ReactWrapper, ShallowWrapper, mount, shallow } from "enzyme";
import { LoginComponent } from "./index";
import { BrowserRouter, Router } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { mockStore } from "../../../__mocks/redux";
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
}));

// const mockStore = configureMockStore();
// const store = mockStore({});

describe("LoginComponent", () => {
  let wrapper: ReactWrapper | ShallowWrapper;
  const mockDispatch = jest.fn();
  jest.mock("react-redux", () => ({
    useDispatch: () => mockDispatch,
  }));

  beforeEach(() => {
    wrapper = shallow(
      <Provider store={mockStore}>
        <BrowserRouter>
          <LoginComponent />
        </BrowserRouter>
      </Provider>
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders without crashing", () => {
    //console.log(wrapper.debug())
    expect(wrapper.exists()).toBeTruthy();
  });

  // it("ensure wrapper to match snapshot", () => {
  //   expect(wrapper).toMatchSnapshot();
  // });

  // it("Should have two input fields and two buttons", () => {
  //   const form = wrapper.find("#login-form");
  //   const usernameInput = wrapper.find(`#login-username input`); // Find the input element inside #login-username
  //   const passwordInput = wrapper.find(`#login-password input`); // Find the input element inside #login-password

  //   const loginButton = wrapper.find("#login-btn button");
  //   const registerButton = wrapper.find("#signup-btn button");
  //   expect(usernameInput).toHaveLength(1);
  //   expect(form).toHaveLength(1);
  //   expect(passwordInput).toHaveLength(1);
  //   expect(loginButton).toHaveLength(1);
  //   expect(registerButton).toHaveLength(1);
  // });

 

  // it("successful login", () => {
  //   const form = wrapper.find("#login-form");
  //   form.simulate("submit");
  //   expect(mockDispatch).not.toHaveBeenCalled();
  // });

  // it("handles login failure", () => {
  //   const form = wrapper.find("#login-form form");
  //   const usernameInput = wrapper.find(`#login-username input`); // Find the input element inside #login-username
  //   const passwordInput = wrapper.find(`#login-password input`); // Find the input element inside #login-password

  //   usernameInput.simulate("change", {
  //     target: { name: "username", value: "testuser" },
  //   });
  //   passwordInput.simulate("change", {
  //     target: { name: "password", value: "invalidpassword" },
  //   });

  //   form.simulate("submit");
  //   expect(mockDispatch).not.toHaveBeenCalled(); // Ensure dispatch is not called on failure
  // });
});
