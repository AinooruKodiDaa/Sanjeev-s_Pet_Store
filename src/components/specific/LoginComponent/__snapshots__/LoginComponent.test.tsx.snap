// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent ensure wrapper to match snapshot 1`] = `ReactWrapper {}`;
